cms/djangoapps/contentstore/courseware_index.py

```diff
@@ -143,6 +145,16 @@ def index_item(item, skip_index=False):
             if not item_index_dictionary and not item.has_children:
                 return
 
+            if groups_usage_info:
+                item.content_groups = []
+                for name, group in groups_usage_info:
```

Comment:
This seems like a strange double loop in here. It feels like we'd be better off twisting the data within groups_usage_info so that we can find matches for this item more efficiently.

Seems like we could build a dict with keys based upon `unicode(module['usage_key_string'])` that would return the array of associated content groups.. then lines 148-153 could become:
`item_content_groups = groups_usage_info.get(item.location, []) if groups_usage_info and else []`

and then we can always refer to item_content_groups without having to use `getattr...` so often